sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -143,7 +143,7 @@ trait ProgressReporter extends Logging {
   }
 
   /** Finalizes the query progress and adds it to list of recent status updates. */
-  protected def finishTrigger(hasNewData: Boolean): Unit = {
+  protected def finishTrigger(hasNewData: Boolean, hasExecuted: Boolean): Unit = {
```

Comment:
Add scala docs to explain the difference between these two flags. This is pretty confusing, and easy mix up when passing the two booleans.